octoprint_mrbeam/software_update_information.py

```diff
@@ -19,72 +24,197 @@
+            ) as f:
+                try:
+                    update_info = json.load(f)
+                    _logger.debug("serverfile_info %s", serverfile_info)
+                    _logger.debug(
+                        "version compare %s - %s",
+                        serverfile_verson,
+                        update_info["version"],
+                    )
+                    if serverfile_verson > update_info["version"]:
+                        newfile = True
+                        _logger.info(
+                            "update local file from server - %s -> %s",
+                            serverfile_verson,
+                            update_info["version"],
+                        )
+
+                except ValueError:
+                    newfile = True
+                    _logger.error(
+                        "there is a wrong configured local config file - override local file with server file"
+                    )
+                except KeyError:
+                    localfilemissing = True
+                    _logger.exception("there is a keyerror in the local file")
+
+                #if local config file invalid override by server
+                if newfile or localfilemissing:
+                    _logger.debug("override local file")
+                    # override local file
+                    f.seek(0)
+                    f.write(serverfile_request.content)
+                    f.truncate()
+                    returncode = self.CLOUD_FILE
+
+        # if server invalid use local file
+        if not serverfile_request:
+            servererror = True
+            returncode = self.LOCAL_FILE
+
+        # if both invalid use repo file
+        if servererror and localfilemissing:
+            _logger.info(
+                "fallback use local default config file /files/software_update/update_info.json"
+            )
+            try:
+                copy(
+                    join(
+                        dirname(realpath(__file__)),
+                        "files/software_update/update_info.json",
+                    ),
+                    get_sw_update_file_path(self._plugin),
+                )
+                newfile = True
+                returncode = self.REPO_FILE
+            except IOError as e:
+                _logger.error(
+                    "not even fallback version available: %s",
+                    e,
+                )
+
+        #if file changed, reload update info
+        if newfile:
+            _logger.info("new file => set info")
+
+            # inform SoftwareUpdate Pluging about new config
+            sw_update_plugin = self._plugin._plugin_manager.get_plugin_info(
+                "softwareupdate"
+            ).implementation
+            sw_update_plugin._refresh_configured_checks = True
+            sw_update_plugin._version_cache = dict()
+            sw_update_plugin._version_cache_dirty = True
+        return returncode
+
+@logme(output=True)
```

Comment:
Do you still need it?

Reply:
no the logme can be removed before the merge, I only needed it for debuging